lms/services/vitalsource/_client.py

```diff
@@ -0,0 +1,94 @@
+from marshmallow import EXCLUDE, Schema, fields
+
+from lms.services.exceptions import ExternalRequestError
+from lms.services.http import HTTPService
+from lms.services.vitalsource.model import VSBookLocation
+from lms.validation._base import RequestsResponseSchema
+
+
+class VitalSourceClient:
+    """
+    A client for making individual calls to VitalSource API.
+
+    See: https://developer.vitalsource.com/hc/en-us/categories/360001974433
+    """
+
+    def __init__(self, api_key: str):
+        """
+        Initialise a client object.
+
+        :param api_key: Key for VitalSource API
+        :raises ValueError: If credentials are invalid
+        """
+        if not api_key:
+            raise ValueError("VitalSource credentials are missing")
+
+        self._http_service = HTTPService()
+        self._http_service.session.headers = {
+            "X-VitalSource-API-Key": api_key,
+            "Accept": "application/json",
+        }
+
+    class _BookInfoSchema(RequestsResponseSchema):
```

Comment:
I personally don't like the inlined schemas. I'd prefer them to be at the top of the file, a bit in between being in another file and inlined.

For me that makes it easier to read the class but this is probably 100% subjective.

Reply:
I've popped them at the bottom. I like the localism of it (everything you need to read is right there), but you're right it is an eyesore.